components/hab/src/cli.rs

```diff
@@ -691,13 +691,17 @@ fn sub_pkg_build() -> App<'static, 'static> {
     );
     // Only a truly native/local Studio can be reused--the Docker implementation will always be
     // ephemeral
-    if cfg!(target_os = "linux") {
-        sub.arg(
+    if cfg!(not(target_os = "macos")) {
```

Comment:
This is kind of tricky because what we're really trying to say is, "show the reuse option if the platform supports native studios", right? The fact that that's equivalent to not Mac OS among our current supported platforms is kind of coincidental. If we added support for arm or solaris down the line, this wouldn't necessarily be the right answer. I'd suggest a helper like `if native_studio_support()` or something. And behind it can be doing a check like `cfg!(linux) || cfg!(windows)`.

Also, the args are sort of wrong because `--reuse` and `--docker` are mutually exclusive on any platform and there's nothing indicating that. But that's an existing problem, so maybe this isn't the place to fix it.

Reply:
great suggestion on the helper method. I would rather tackle the reuse/docker issue in a separate pr.